app/javascript/components/forms/SurveyForm.vue

```diff
@@ -114,74 +110,61 @@ export default {
     axiosCall() {
       axios(this.formData.config.action, this.options)
         .then((response) => {
-          this.isLastPage && this.axiosDone ? this.hasNoErrors = true : this.hasNoErrors = false
-          Turbolinks.visit(response.data.redirect_path);
+          if (response.data.redirect_path) {
+            // preferred to turbolink so JQery reloads on the commitment form
+            window.location.replace(response.data.redirect_path)
+          }
         })
         .catch((error) => {
-          console.log('FAILED!', error.response.data.errors);
-          this.hasNoErrors = false
-          this.errors = error.response.data.errors;
+          if (error.response) {
+            this.errors = error.response.data.errors;
+          }
         });
     },
 
     complete() {
-      this.survey.completeLastPage();
+      this.onComplete(this.survey)
```

Comment:
Is there a benefit to doing it this way? We set up a callback for the hook that `completeLastPage` triggers  on line 82, so it seems more conventional to use the survey method and trigger the hook to trigger the callback rather than skipping the hook and going straight to the callback?

Reply:
Unfortunately completeLastPage triggers doComplete in SurveyJs but not the onComplete event which we want to fire to amend the data object.